src/core/server/http/http_config.test.ts

```diff
@@ -54,6 +54,37 @@ test('throws if invalid hostname', () => {
   expect(() => httpSchema.validate(obj)).toThrowErrorMatchingSnapshot();
 });
 
+describe('acceptOpaqueIdFrom', () => {
+  test('accepts valid ip addresses', () => {
+    const {
+      requestOpaqueId: { ipAllowlist },
+    } = config.schema.validate({
+      requestOpaqueId: {
+        ipAllowlist: ['0.0.0.0', '123.123.123.123', '1200:0000:AB00:1234:0000:2552:7777:1313'],
```

Comment:
1/ so it's IP+ optional port, not 'just' IP?

Seeing the check `this.options.requestOpaqueId.ipAllowlist.includes(this.socket.remoteAddress!)`, allowing port is probably error prone, as `remoteAddress` is only the IP.

```ts
 /** remote client IP address. */
    remoteAddress: string;
    /** remote client port. */
    remotePort: string;
```

2/ do we want to allow IP ranges?

Reply:
JFYI: Joi also supports `cidr`

Reply:
1/ No ports here, just IPv4 and IPv6.

2/ Possibly. @jportner do you know? I know you mentioned this wasn't required for the MVP but it wasn't a lot of work to add it. If there are too many unknown requirements right now, I can just remove.

Reply:
> No ports here, just IPv4 and IPv6.

My bad, thought I saw a IPv4 with a `:port`.

Reply:
> 2/ Possibly. @jportner do you know? I know you mentioned this wasn't required for the MVP but it wasn't a lot of work to add it. If there are too many unknown requirements right now, I can just remove.

I don't think ranges are necessary, up to you.